firmware/src/calc.c

```diff
@@ -10,10 +10,10 @@
 // arr[0] is always most recent sample
 int16_t moving_average(int16_t sample, int16_t arr[MOVING_AVG_FILTER_LEN]) {
     uint8_t i;
-    int32_t sum = 0;
-    for (i = 1; i < MOVING_AVG_FILTER_LEN; i++) {
-        arr[i] = arr[i-1];
+    int32_t sum = arr[0];
+    for (i = 1; i < MOVING_AVG_FILTER_LEN - 1; i++) {
```

Comment:
Are we off-by-one here?

Reply:
Not afaik. 

Because we initialize the sum to contain arr[0], we only need to iterate through MOVING_AVG_FILTER_LEN - 2 elements because we're going to add the new sample to the sum after this for loop. We initialize i = 1 so we need to only loop until MOVING_AVG_FILTER_LEN - 1. 

If MOVING_AVG_FILTER_LEN = 5 and the size of the array is 5 then we add element 0 to sum before the for loop, add element 1, 2, and 3 to the sum during the for loop and then we add the sample after the loop so the sum is constituted of 5 elements. If we didn't have the '-1' then we would be summing over 6 elements.